File: documentation/manual/working/javaGuide/main/dependencyinjection/code/javaguide/di/guice/Module.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package javaguide.di.guice;

import javaguide.di.*;

// #guice-module
import com.google.inject.AbstractModule;
import com.google.inject.name.Names;

public class Module extends AbstractModule {
  protected void configure() {

    bind(Hello.class).annotatedWith(Names.named("en")).to(EnglishHello.class);

    bind(Hello.class).annotatedWith(Names.named("de")).to(GermanHello.class);
  }
}
// #guice-module

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/dev-mode/app/Module.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

import java.io.FileWriter
import java.util.Date

import com.google.inject.AbstractModule
import play.api._

class Module(environment: Environment, configuration: Configuration) extends AbstractModule {

  override def configure() = {
    val writer = new FileWriter(environment.getFile("target/reload.log"), true)
    writer.write(s"${new Date()} - reloaded\n")
    writer.close()

    if (configuration.getOptional[Boolean]("fail").getOrElse(false)) {
      throw new RuntimeException("fail=true")
    }
  }
}
